docs/vscode/lobster/.vscode/launch.json

```diff
@@ -8,10 +8,15 @@
         {
             "name": "Extension",
             "type": "extensionHost",
-            "request": "launch",
-            "args": [
-                "--extensionDevelopmentPath=${workspaceFolder}"
-            ]
+			"request": "launch",
+			"name": "Launch Client",
+			"runtimeExecutable": "${execPath}",
+			"args": ["--extensionDevelopmentPath=${workspaceRoot}"],
+			"outFiles": ["${workspaceRoot}/client/out/**/*.js"],
+			"preLaunchTask": {
+				"type": "npm",
```

Comment:
is this correct? should users of a plugin run `npm` all the time?

Reply:
This is when you are testing the plugin, it runs `npm watch` so to hot update typescript to js.

Reply:
Right, but stuff in this dir is what users copy to their `.vscode` folders. If this stuff is only used during development, then we should have a folder inside `dev/lsp` that holds this stuff, and keep this folder clean for users.

Reply:
Hmm, this is true normally .vscodeignore is used for this. When you publish a extension those files are not included. I can move the actual implementation away but im unsure what to put here then.

Reply:
A `.vscodeignore` is not used by windows file copy afaik ;) Users will end up with lots of dev files they don't use.

You don't need anything here. This is a folder users copy. You can make a similar folder in `dev/lsp` that contains all these files, and work/test from there instead of this folder.

Reply:
But the extension is based of the files in (lets call it) `dev/vscode`. Normaly there runs a process that uses .vscodeignore for this. We can automate this with github actions but then its a commit that updates this after every commit. And manualy updating it sounds also a bad idea. Another solution would remove this folder and have a artifact be prepared.

Reply:
We can either do it manually, or in CI, or maybe better yet, this should be uploaded to the VSCode extensions repository so we can just tell users to get the extension inside VSCode?

I suggest to start with manual for now, then find out better ways later. It is not that much work to run whatever command packages the extension before you commit any changes to the code. Just add a .bat/.sh

Regardless, user and dev files should not be mixed.